index/score.go

```diff
@@ -262,6 +262,14 @@ func (p *contentProvider) calculateTermFrequency(cands []*candidateMatch) map[st
 		}
 	}
 
+	// If a file is a test, generated, etc., then cut its term frequency in half (rounded up). The BM25F interpretation
+	// is that this data lives in a separate 'field' that is half the priority of regular content.
+	if lowPriority {
+		for term := range termFreqs {
+			termFreqs[term] = termFreqs[term]/2 + 1
```

Comment:
I also tried down-weighting by 5 (to match the constant 5 above for filenames and symbols). This also worked well in testing, and had the same effect on evals. However, I noticed that when you are explicitly searching for a test, like `ranking test`, this is a bit aggressive and makes it harder to find the right file.

I am totally open to changing it though. When I looked at the "mentions dataset" in SPLF-864, I found that people only @-mention test files 4% of the time (compared to 92% for "regular" files!) so it may be rare to search for tests.

Reply:
I like this approach, it's very simple. While testing it locally, I noticed the impact, but it was very subtle. Why do we manipulate the term frequencies instead of the final score? Test cases might have a very high repetition of symbols and halving the frequency might not be enough? There is also the issue that test files are often much shorter. Either way, for BM25, I think this approach is better than the tiebreaker.

Reply:
@stefanhengl could you try testing with 1/5 downweighting too, and give your feedback on whether that feels better?